examples/bri-3/src/bri/zkp/agents/proof.agent.ts

```diff
@@ -0,0 +1,49 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import {
+  INVALID_PROOF_INPUT,
+  NOT_FOUND_ERR_MESSAGE,
+} from '../api/err.messages';
+import { Proof } from '../models/proof';
+import { ProofStorageAgent } from './proofStorage.agent';
+import { BpiAccount } from '../../identity/bpiAccounts/models/bpiAccount';
+
+@Injectable()
+export class ProofAgent {
+  constructor(private storageAgent: ProofStorageAgent) {}
+
+  public throwIfProofInputInvalid(document: string): boolean {
+    if (document == '') {
+      throw new BadRequestException(INVALID_PROOF_INPUT);
+    }
+    return true;
+  }
+
+  public createNewProof(
+    id: string,
+    owner: BpiAccount,
```

Comment:
should this not be `BpiSubjectAccount`? also do we not also need the `BpiAccount` to be able to reference the state that should be advanced by the document as the input?

Reply:
Agree on this being the BpiSubjectAccount. @Therecanbeonlyone1969 The state object only needs to be associated with an existing BpiAccount, correct? if the BpiAccount is like an EOA, what reference to the state should it need to make?